### PR TITLE
Remove literals passed to `device_uvector::set_element_async`

### PR DESCRIPTION
**Pull request by harrism**
After rapidsai/rmm#725 is merged, this PR updates cuspatial to eliminate passing literal values to device_uvector::set_element_async.

Companion PR to rapidsai/cuspatial#367

### CODE REVIEW

**Review by seunghwak — APPROVED** [review]
Thanks for the update!

**Review by afender — APPROVED** [review]
Thanks, Mark.

**Review by aschaffer — APPROVED** [review]

**Comment by harrism** [comment]
codecov failing to connect in CI?

**Comment by harrism** [comment]
rerun tests

**Comment by harrism** [comment]
@BradReesWork this should be safe to merge and good to get it in before RMM #725 merges. Why the DO NOT MERGE label?

**Comment by BradReesWork** [comment]
@harrism I marked it as blocked based on your comment "After rapidsai/rmm#725 is merged,"  If there is not an issue for cuGraph, then I'll merge it now

**Comment by BradReesWork** [comment]
@gpucibot merge

**Comment by harrism** [comment]
Ah sorry, I meant RMM 725 would break cuGraph build if merged, but this was safe before and after 725.

**Pull request merged**
